consult-flymake.el

```diff
@@ -70,14 +70,16 @@ RAW-DIAGS should be a list of diagnostics as returned from `flymake-diagnostics'
                    'consult--candidate marker
                    'consult--type narrow))
      ;; Sort first by position and then by the name of the buffer.
-     (sort
-      (sort diags
-            (pcase-lambda (`(,b1 _ ,t1 _ ,m1 _) `(,b2 _ ,t2 _ ,m2 _))
-              (let ((s1 (flymake--severity t1))
-                    (s2 (flymake--severity t2)))
-                (or (> s1 s2) (and (= s1 s2) (< m1 m2))))))
-      (pcase-lambda (`(,b1 _ _ _ _ _) `(,b2 _ _ _ _ _))
-        (string-lessp b1 b2))))))
+     (sort diags
+           (pcase-lambda (`(,b1 _ ,t1 _ ,m1 _) `(,b2 _ ,t2 _ ,m2 _))
+             (let ((s1 (flymake--severity t1))
+                   (s2 (flymake--severity t2)))
+               (or
+                (> s1 s2)
+                (string-greaterp b1 b2)
+                (and (= s1 s2)
+                     (string-lessp b1 b2)
+                     (< m1 m2)))))))))
```

Comment:
This doesn't look correct.

Reply:
What do you want to achieve?

Sort first by buffer name, then severity, then position?

Reply:
This is still wrong. I will check tomorrow. Please think about this more thoroughly.